packages/cardpay-sdk/sdk/prepaid-card-market-v-2/base.ts

```diff
@@ -0,0 +1,446 @@
+import Web3 from 'web3';
+import { AbiItem, fromWei, toWei } from 'web3-utils';
+import PrepaidCardMarketV2ABI from '../../contracts/abi/v0.9.0/prepaid-card-market-v-2';
+import BN from 'bn.js';
+import { ContractOptions } from 'web3-eth-contract';
+import { isTransactionHash, TransactionOptions, waitForTransactionConsistency } from '../utils/general-utils';
+import {
+  EventABI,
+  executeSend,
+  executeSendWithRateLock,
+  executeTransaction,
+  gasEstimate,
+  getNextNonceFromEstimate,
+  getParamsFromEvent,
+  getSendPayload,
+  GnosisExecTx,
+  Operation,
+  SendPayload,
+} from '../utils/safe-utils';
+import { Signature, signPrepaidCardSendTx, signSafeTx } from '../utils/signing-utils';
+
+import { ZERO_ADDRESS } from '../constants';
+import { Signer } from 'ethers';
+import ERC677ABI from '../../contracts/abi/erc-677';
+import { SuccessfulTransactionReceipt } from '../utils/successful-transaction-receipt';
+import { getAddress } from '../..';
+
+export default class PrepaidCardMarketV2 {
+  constructor(private layer2Web3: Web3, private layer2Signer?: Signer) {}
+
+  async isPaused(marketAddress?: string): Promise<boolean> {
+    marketAddress = marketAddress ?? (await getAddress('prepaidCardMarketV2', this.layer2Web3));
+    let contract = new this.layer2Web3.eth.Contract(PrepaidCardMarketV2ABI as AbiItem[], marketAddress);
+    return await contract.methods.paused().call();
+  }
+
+  private async getAddTokensPayload(issuerAddress: string, tokenAddress: string, amount: BN): Promise<string> {
+    let token = new this.layer2Web3.eth.Contract(ERC677ABI as AbiItem[], tokenAddress);
+    let prepaidCardMarketV2Address = await getAddress('prepaidCardMarketV2', this.layer2Web3);
+    let data = this.layer2Web3.eth.abi.encodeParameters(['address'], [issuerAddress]);
+    return token.methods.transferAndCall(prepaidCardMarketV2Address, amount, data).encodeABI();
+  }
+
+  async addTokens(txnHash: string): Promise<SuccessfulTransactionReceipt>;
+  async addTokens(
```

Comment:
Hos is this different than the normal Safe,sendTokens API? https://github.com/cardstack/cardstack/blob/main/packages/cardpay-sdk/sdk/safes.ts#L313 it seems like duplicate code.

Reply:
It's different where we add the issuer's address to the payload: https://github.com/cardstack/cardstack/pull/2979/files/85a162b4d486bad1f5a72e90d9cb2812e98d1a24#diff-3e7b707c9d90125a88f86bb8d8c85c1f4514dbb328c263bf955d7124876ba443R40

Do you think it's fine to have similar code in this place but with this difference, or should we refactor sendTokens to support adding params to the `data`?